src/toolkit/res_buf.h

```diff
@@ -163,6 +163,35 @@ class ResBuf {
     return Pop(qty);
   }
 
+  /// Pops the specified amount of material, re-packed into the the 
+  /// package type provided. Returns any remaining material that doesn't
+  /// fit in a package per the packaging strategy and limits as a 
+  /// separate vector with default packaging
+  std::vector<typename T::Ptr> PopPackaged(double qty, Package::Ptr pkg) {
+    typename T::Ptr r = Pop(qty);
```

Comment:
Conceptual: this `Pop` will squash resources to get this `qty`.  What if it's already packaged in this `pkg` type and we could just `Pop` multiple items with `PopN`?  Is there a way to test that?

Reply:
What if we always `PopN` to get enough `qty` and then have a `repackage` method that takes some number of packaged materials and puts them into a different number of different packaged materials?  The `repackage` method could be something that we look to reuse in other places